.github/labeler.yml

```diff
@@ -0,0 +1,36 @@
+build:
+- default.nix
+- CMakeLists.txt
+- build/*
+- build/**/*
+- cmake/*
+- cmake/**/*
+
+code quality:
+- src/test/*
+- src/test/**/*
+- .clang-format
+- .codespell
+- .eslint*
+- .flake8
+- .pre-commit-config.yaml
+- pyproject.toml
+
+controllers:
+- res/controllers/*
+
+library:
+- src/library/*
+- src/library/**/*
+
+skins:
+- res/skins/*
+- res/skins/**/*
+
+ui:
+- src/*.ui
```

Comment:
Weird that we need to repeat all patterns for a recursive match including the top-level directory.

Reply:
Maybe we don't. The docs are not really clear and the common examples have both: https://github.com/actions/labeler#common-examples

I can remove the duplicates, and we read them in case the labeler doesn't work as intended.

Reply:
The pattern `src/**.ui` matches all .ui files in src and its subdirectories. `**` matches all characters including `/` while `*` does not match `/`.